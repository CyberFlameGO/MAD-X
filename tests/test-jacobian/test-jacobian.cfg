1-7     *     skip         # head
77      *     skip         # tail

# first match
35-36   1-2   rel=1e-12
37      3     abs=1e-21
39      1     rel=1e-12

# first match summary
50      1     abs=1e-12
55      1     abs=1e-12
59      1     abs=1e-21

# first variable
66      1     abs=1e-12
70      1     abs=1e-21

