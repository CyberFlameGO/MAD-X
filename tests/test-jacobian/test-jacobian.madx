x1=0.1;
match,use_macro;
vary,name=x1;
vary,name=x2;
m1: macro={;};
constraint,expr= x1+3*x2=1;
constraint,expr= x1+1*x2=1;
jacobian,calls=20,tolerance=1.e-21;
endmatch;

