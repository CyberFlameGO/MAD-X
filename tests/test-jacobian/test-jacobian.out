
  +++++++++++++++++++++++++++++++++++++++++++
  +    MAD-X 5.00.16  (64 bit, Darwin)      +
  +    DEBUG Version - use at own risk!     +
  + Code Modification Date: 2012.04.05      +
  + Execution Time Stamp: 19.06.12 16.50.08 +
  +++++++++++++++++++++++++++++++++++++++++++
x1=0.1;

match,use_macro;

vary,name=x1;

vary,name=x2;

m1: macro={;};

constraint,expr= x1+3*x2=1;

constraint,expr= x1+1*x2=1;

jacobian,calls=20,tolerance=1.e-21;

number of variables:    2
user given constraints: 0
total constraints:      2

START JACOBIAN:

 JACOBIAN Strategy =           3
Initial Penalty Function =   0.1620000000000000E+01


 Solve system with            2 con,           2 var
 Rank             2   Condition number   5.82842712474619     
 Step length  0.899999999999851     
call:     2 Dx =   0.90000000E+00  Penalty function =  0.4463902367581577E-23
 ++++++++++ JACOBIAN ended: converged successfully
 Final difference norm:  0.948683298049726     
endmatch;


MATCH SUMMARY

--------------------------------------------------------------------
macro: m1                  
  constraint: x1+3*x2=1                               
  values:     +1.00000e+00=+1.00000e+00
  weight:     +1.00000e+00
  penalty:    -1.49403e-12

  constraint: x1+1*x2=1                               
  values:     +1.00000e+00=+1.00000e+00
  weight:     +1.00000e+00
  penalty:    -1.49392e-12



Final Penalty Function =   4.46390237e-24



Variable                 Final Value  Initial Value Lower Limit  Upper Limit 
--------------------------------------------------------------------------------
x1                        1.00000e+00  1.00000e-01 -1.00000e+20  1.00000e+20
x2                       -3.92523e-17  0.00000e+00 -1.00000e+20  1.00000e+20

END MATCH SUMMARY

VARIABLE "TAR" SET TO   4.46390237e-24



  Number of warnings: 0

  ++++++++++++++++++++++++++++++++++++++++++++
  + MAD-X 5.00.16 64 finished normally +
  ++++++++++++++++++++++++++++++++++++++++++++
